File: src/main/java/org/ebics/client/order/AbstractEbicsDownloadOrder.java
package org.ebics.client.order;

import java.util.Date;
import java.util.Map;

public abstract class AbstractEbicsDownloadOrder extends EbicsOrder {

    /**
     * Download order
     * @param adminOrderType the EBICS admin order type
     * @param startDate start date (for historical downloads only)
     * @param endDate end date (for historical downloads only)
     */
    public AbstractEbicsDownloadOrder(EbicsAdminOrderType adminOrderType, Date startDate, Date endDate, Map<String, String> params) {
        super(adminOrderType, params);
        this.startDate = startDate;
        this.endDate = endDate;
    }

    public Date getStartDate() {
        return startDate;
    }

    public Date getEndDate() {
        return endDate;
    }

    private Date startDate;
    private Date endDate;
}

File: ebics-java-lib/src/main/kotlin/org/ebics/client/order/AbstractEbicsDownloadOrder.kt
package org.ebics.client.order

import java.util.*

/**
 * Download order
 * @param adminOrderType the EBICS admin order type
 * @param startDate start date (for historical downloads only)
 * @param endDate end date (for historical downloads only)
 */
abstract class AbstractEbicsDownloadOrder
    (adminOrderType: EbicsAdminOrderType, val startDate: Date?, val endDate: Date?, params: Map<String, String>) :
    EbicsOrder(
        adminOrderType, params
    )